mypy/checker.py

```diff
@@ -2883,7 +2883,45 @@ def conditional_type_map(expr: Expression,
         return {}, {}
 
 
-def partition_by_callable(type: Type) -> Tuple[List[Type], List[Type]]:
+def intersect_instance_callable(type: Instance, callable_type: CallableType) -> Type:
```

Comment:
Style nit: we generally prefer `typ` instead of `type` since `type` is a builtin :-( This doesn't apply to attribute access, since it's clear that `foo.type` doesn't refer to the builtin.